homepage/tests/sign_in_tests.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+from django.contrib.auth.models import User
+from django.contrib.auth import authenticate
+from pytest_django.asserts import assertTemplateUsed
+
+
+@pytest.mark.django_db
+class TestSignIp:
+
+    @pytest.fixture
+    def signed_up_details(self):
+        User.objects.create_user(username='valid_username', email='valid@email.com', password='pw123123')
+        return ['valid_username', 'pw123123']
+
+    @pytest.fixture
+    def not_signed_up_details(self):
+        return ['not_valid_username', 'pw123123']
+
+    # ------------------------Back-End testing------------------------ #
+
+    def test_signin_invalid(self, not_signed_up_details):
+        user = authenticate(username=not_signed_up_details[0], password=not_signed_up_details[1])
+        assert user is None
+
+    def test_signin_valid(self, signed_up_details):
+        user = authenticate(username=signed_up_details[0], password=signed_up_details[1])
+        assert user
+
+    # ------------------------Front-End testing------------------------ #
+
+    def test_post_valid_signIn_with_client(self, client, signed_up_details):
+        response = client.post('/users/sign_in/', data={
+            'username': signed_up_details[0], 'password': signed_up_details[1]})
+
+        assert response.status_code == 302

```

Comment:
Maybe you can check that after authentication, the user is redirected to the landing page.
Like in the presentation:
![image](https://user-images.githubusercontent.com/57869309/117531639-2a321500-afec-11eb-89d9-b8c52b9b71f4.png)

Reply:
Added a check for redirection to homepage after successful sign-in